### PR TITLE
Fix a crash in NetworkAvailability

### PR DESCRIPTION
**Pull request by FabianDevel**
Fix MAIL-ANDROID-EX5

There is a crash because of a concurrent read and write in NetworkAvailability

### CODE REVIEW

File: src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt [hunk 1]
```diff
@@ -32,24 +32,35 @@ import kotlinx.coroutines.channels.awaitClose
...
 
     val isNetworkAvailable: Flow<Boolean> = callbackFlow {
         val networks = mutableListOf<Network>()
 
         val callback = object : NetworkCallback() {
 
             override fun onAvailable(network: Network) {
-                networks.add(network)
-                sendNetworkAvailability(networks)
+                launch {
```
Comment (KevinBoulongne) [inline src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt:49]:
```suggestion
                launch(ioDispatcher) {
```

File: src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt [hunk 2]
```diff
@@ -32,24 +32,35 @@ import kotlinx.coroutines.channels.awaitClose
...
+                        networks.add(network)
+                        sendNetworkAvailability(networks)
+                    }
+                }
             }
 
             override fun onLost(network: Network) {
-                networks.remove(network)
-                sendNetworkAvailability(networks)
+                launch {
```
Comment (KevinBoulongne) [inline src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt:58]:
```suggestion
                launch(ioDispatcher) {
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt (hunk 1 above)
Comment (FabianDevel) [reply src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt:49]: It is on purpose that it runs on the default. 
As we're already in a flow, the default dispatcher will just be launched in background and do its thing.
The io has a special algorithm that wakes it after external read/write (explanation from sirambd), we don't want this here

File: src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt (hunk 2 above)
Comment (FabianDevel) [reply src/main/java/com/infomaniak/lib/core/networking/NetworkAvailability.kt:58]: same as above

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**